bindings/py/cpp_src/plugin/PyBindRegion.cpp

```diff
@@ -207,6 +207,12 @@ namespace py = pybind11;
             continue;
           if (!ns->parameters.contains(key))
             NTA_THROW << "Parameter '" << key << "' is not expected for this Region.";
+            
+          // Prevent parameters with ReadOnlyAccess from being used
+          // to initialize a region.  They must have CreateAccess or ReadWriteAccess.
```

Comment:
Please refresh my memory, why can't a RO param be used?

Reply:
The idea is that some parameters are status points generated by the algorithm and not part of the arguments passed to the algorithm during creation.  These are the `ReadOnly` parameters.
Parameters that are used as creation parameters and then are read only after would be `Create` parameters.  Some parameters can be modified after creation and those are the `ReadWrite` parameters.

But 'Why'?  Well, this was the way that someone in the past decided it should be on the C++ side and it seemed reasonable to make it the same on the Python side.  The advantage is that we don't include the ReadOnly parameters in the set of parameters used to create an algorithm.  On the python side it is probably ignored but this would be cleaner.